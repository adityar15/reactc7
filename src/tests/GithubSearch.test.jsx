import React from 'react'
import renderer from 'react-test-renderer'


import { expect, test } from 'vitest'
import GithubSearchProvider from '../pages/GithubSearch';

// tools used
// react-test-renderer happy-dom jsdom 

// function toJson(component) {
//   const result = component.toJSON()
//   expect(result).toBeDefined()
//   expect(result).not.toBeInstanceOf(Array)
//   return result
// }
test('Github search component', () => {

  const component = renderer.create(<GithubSearchProvider title="Adi" />);
  let tree = component.toJSON()
  console.log("to json", tree)
  expect(tree).toMatchSnapshot();


  // If you need to test different states or props, you can add more test cases here.
});