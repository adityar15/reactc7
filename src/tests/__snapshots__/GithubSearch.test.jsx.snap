// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Github search component 1`] = `
<div
  className="max-w-2xl mx-auto p-2"
>
  <form
    className="flex items-center gap-1"
    onSubmit={[Function]}
  >
    <input
      className="border border-gray-300 rounded p-2 flex-grow"
      name="username"
      onInput={[Function]}
      placeholder="Enter your Github username"
      type="search"
    />
    <button
      className="bg-blue-500 text-white px-5 py-2 rounded"
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;

exports[`Github search component test to see if it renders correctly on input 1`] = `
<div
  className="max-w-2xl mx-auto p-2"
>
  <form
    className="flex items-center gap-1"
    onSubmit={[Function]}
  >
    <input
      className="border border-gray-300 rounded p-2 flex-grow"
      name="username"
      onInput={[Function]}
      placeholder="Enter your Github username"
      type="search"
    />
    <button
      className="bg-blue-500 text-white px-5 py-2 rounded"
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;
